### PR TITLE
Add VERSIONINFO resource

### PR DESCRIPTION
**Pull request by nkh0472**
Display version info and file name under Windows.

Tested through latest Mingw-w64.

### CODE REVIEW

File: fdkaac.rc [hunk 1]
```diff
@@ -1,3 +1,37 @@
 #include <winresrc.h>
 
 CREATEPROCESS_MANIFEST_RESOURCE_ID RT_MANIFEST "fdkaac.manifest"
+
+#define VER_VERSION 1,0,2,0
+//The version parameter is two 32-bit integers, defined by four 16-bit integers.
+#define VER_VERSION_STR "1.0.2"
+#define VER_LEGALCOPYRIGHT_STR "(c) nu774 2013-2021"
```
Comment (nkh0472) [inline fdkaac.rc:8]: I'm not sure if you like this copyright notice, you can modify it or delete this line and line 26 if you don't like it.

**Comment by nu774** [comment]
Thanks for the patch.
I think it would be better if version number is automatically picked up from version.h.
Is it possible?

File: fdkaac.rc [hunk 2]
```diff
@@ -1,3 +1,33 @@
...
+FILETYPE 0x1
+//VFT_APP 0x1 File contains an application.
+
+{
+BLOCK "StringFileInfo"
+{
+	BLOCK "040904E4"
+	{
+		VALUE "FileDescription", "Command line frontend for libfdk-aac encoder"
+		VALUE "FileVersion", VER_VERSION_STR
```
Comment (nkh0472) [inline fdkaac.rc:21]: I don't know why I can't use that variable `fdkaac_version`. 
```suggestion
		VALUE "FileVersion", fdkaac_version
```
That will cause 
```
$ windres fdkaac.rc -o fdkaac.o
E:\Etc\HUGE\msys64\mingw64\bin\windres.exe: fdkaac.rc:21: syntax error
```

**Comment by nkh0472** [comment]
> I think it would be better if version number is automatically picked up from version.h.

Now the definition of the version number has been moved to `version.h`.

**Comment by nu774** [comment]
Did you try re-running autoreconf ?
It now fails because configure.ac also picks up version information from version.h, and now it is different.
The first lines of configure.ac has to be edited to something like this.
```
m4_define([VERSION_STR],m4_esyscmd([grep VER_VERSION_STR version.h]))
changequote({{,}})dnl
m4_define({{XX_VERSION}},m4_bregexp(VERSION_STR,{{.*VER_VERSION_STR "\(.+\).0".*}},{{\1}}))
```
Also, I'm skeptical of the last two commits which add explicit NUL at the end of the string.
Are they *really* needed? Does that make difference on the generated file?

**Comment by nkh0472** [comment]
> Did you try re-running autoreconf ?
> It now fails because configure.ac also picks up version information from version.h, and now it is different.
> The first lines of configure.ac has to be edited to something like this.

For the last four commit, I only tested `make` and the output of the complied fdkaac.exe.

> Also, I'm skeptical of the last two commits which add explicit NUL at the end of the string.
> Are they _really_ needed? Does that make difference on the generated file?

I only saw such a sentence on [a Chinese blog page](http://www.cppblog.com/luqingfei/archive/2010/09/21/127239.html) today. Although the author did not give a reference, but in [Microsoft's official example](https://docs.microsoft.com/en-us/windows/win32/menurc/versioninfo-resource), all of the strings are written like this. So I think it's better to keep the writing consistent with the official documents.

Although there is no `\0` as the end in this commit before, and the file properties can be displayed normally in my Windows 10; But I'm worried about whether this will cause compatibility or other problems on lower version systems or another compiler.

**Comment by nu774** [comment]
In C programming language, any string literal is terminated by a NUL, so you generally don't need them unless you need double NULs.
If you are unsure, you should at least check if output of resource compiler (windres) is actually different or not.
To begin with, don't push code if you are unsure. This is not your own repo.

**Comment by nkh0472** [comment]
> If you are unsure, you should at least check if output of resource compiler (windres) is actually different or not.

They are the same. But I still think it's better to follow the example of the official document.

> To begin with, don't push code if you are unsure. This is not your own repo.

I'm very sorry for the problem I caused. I will remember your advice.

**Pull request closed without merging**